packages/js/product-editor/src/components/button-with-dropdown-menu/index.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import { chevronDown } from '@wordpress/icons';
+import { Button, DropdownMenu, Flex, FlexItem } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import type { ButtonWithDropdownMenuProps } from './types';
+
+export const ButtonWithDropdownMenu: React.FC<
+	ButtonWithDropdownMenuProps
+> = ( {
+	text,
+	dropdownButtonLabel = __( 'More options', 'woocommerce' ),
+	onButtonClick = () => {},
+	controls = [],
+	variant = 'primary',
+	defaultOpen = false,
+	popoverProps: {
+		placement = 'bottom-end',
+		position = 'bottom left left',
+		offset = 0,
+	} = {
+		placement: 'bottom-end',
+		position: 'bottom left left',
+		offset: 0,
+	},
```

Comment:
I wonder if it's worth adding a `className` prop to provide more flexibility. What do you think about that?

Reply:
SGTM! Let me do it in a follow-up/ 🙇‍♂️

Reply:
https://github.com/woocommerce/woocommerce/pull/43136